torch/csrc/jit/codegen/cuda/kernel_ir.cpp

```diff
@@ -539,6 +539,7 @@ Val* newLogicExpr(BinaryOpType op_type, Val* lhs, Val* rhs) {
 } // namespace
 
 Val* lowerValue(const Val* val) {
+  TORCH_INTERNAL_ASSERT(!kir::isLoweredVal(val), val, " is already lowered.");
```

Comment:
technically this is not needed since Fusion::lowerValue would assert otherwise. we can keep it if it helps though.

Reply:
I didn't like the cryptic dispatch error.